src/AST/Predicates.dfy

```diff
@@ -102,6 +126,7 @@ abstract module Base {
           && All_Expr(body, P)
         case If(cond, thn, els) =>
           All_Expr(cond, P) && All_Expr(thn, P) && All_Expr(els, P)
+        case Unsupported(_, _) => true
```

Comment:
I had this idea that subexpressions of unsupported expressions would be handled differently, and not using the normal traversals, thinking that it would be awkward to get all the proofs to go through over incomplete terms. Turns out it's no problem at all, so I've moved toward treating them symmetrically with all other expressions.